ProcessLib/TES/TESOGS5MaterialModels.h

```diff
@@ -0,0 +1,399 @@
+
+struct FluidViscosityH2O
+{
+	static double get(double rho, double T)
+	{
+		double my,my_0,my_1;
+		double H[4];
+
+		T = T / 647.096;
+		rho = rho / 322.0;
+
+		H[0] = 1.67752;
+		H[1] = 2.20462;
+		H[2] = 0.6366564;
+		H[3] = -0.241605;
+
+		double h[6][7] = { 0.0 };
+		h[0][0] = 0.520094000;
+		h[1][0] = 0.085089500;
+		h[2][0] = -1.083740000;
+		h[3][0] = -0.289555000;
+		h[0][1] = 0.222531000;
+		h[1][1] = 0.999115000;
+		h[2][1] = 1.887970000;
+		h[3][1] = 1.266130000;
+		h[5][1] = 0.120573000;
+		h[0][2] = -0.281378000;
+		h[1][2] = -0.906851000;
+		h[2][2] = -0.772479000;
+		h[3][2] = -0.489837000;
+		h[4][2] = -0.257040000;
+		h[0][3] = 0.161913000;
+		h[1][3] = 0.257399000;
+		h[0][4] = -0.032537200;
+		h[3][4] = 0.069845200;
+		h[4][5] = 0.008721020;
+		h[3][6] = -0.004356730;
+		h[5][6] = -0.000593264;
+
+		double sum1 = H[0] / mypow<0>(T);
+		sum1 += H[1] / mypow<1>(T);
+		sum1 += H[2] / mypow<2>(T);
+		sum1 += H[3] / mypow<3>(T);
+
+		my_0 = 100 * std::sqrt(T) / sum1;
+
+		double sum2 = inner_loop<0>(rho, T, h);
+		sum2 += inner_loop<1>(rho, T, h);
+		sum2 += inner_loop<2>(rho, T, h);
+		sum2 += inner_loop<3>(rho, T, h);
+		sum2 += inner_loop<4>(rho, T, h);
+		sum2 += inner_loop<5>(rho, T, h);
+
+		my_1 = std::exp(rho * sum2);
+
+		my = (my_0 * my_1) / 1e6;
+		return my;
+	}
+
+private:
+	template <int i>
+	static double inner_loop(const double rho, const double T, const double (&h)[6][7])
+	{
+		const double base = rho - 1.0;
+
+		double sum3 = h[i][0] * mypow<0>(base);
+		sum3 += h[i][1] * mypow<1>(base);
+		sum3 += h[i][2] * mypow<2>(base);
+		sum3 += h[i][3] * mypow<3>(base);
+		sum3 += h[i][4] * mypow<4>(base);
+		sum3 += h[i][5] * mypow<5>(base);
+		sum3 += h[i][6] * mypow<6>(base);
+
+		return mypow<i>(1 / T - 1) * sum3;
+	}
+};
+
+inline double fluid_viscosity(const double p, const double T, const double x)
+{
+	// OGS 5 viscosity model 26
+
+	const double M0 = ProcessLib::TES::M_N2;
+	const double M1 = ProcessLib::TES::M_H2O;
+
+	//reactive component
+	const double x0 = M0*x/(M0*x + M1*(1.0-x)); //mass in mole fraction
+	const double V0 = FluidViscosityH2O::get(M1*p/(GAS_CONST*T), T);
+	//inert component
+	const double x1 = 1.0 - x0;
+	const double V1 = FluidViscosityN2::get(M0*p/(GAS_CONST*T), T);
+
+	const double M0_over_M1 (M1/M0); //reactive over inert
+	const double V0_over_V1 (V0/V1);
+
+	const double phi_12 = mypow<2>(1.0 + std::sqrt(V0_over_V1) * std::pow(1.0/M0_over_M1, 0.25))
+						  / std::sqrt(8.0*(1.0+M0_over_M1));
+
+	const double phi_21 = phi_12 * M0_over_M1 / V0_over_V1;
+
+	return V0*x0 / (x0 + x1 * phi_12)
+			+ V1*x1 / (x1 + x0 * phi_21);
+}
+
+
+struct FluidHeatConductivityN2
+{
+	static double get(double rho, double T)
+	{
+		const double X1 = 0.95185202;
+		const double X2 = 1.0205422;
+
+		const double rho_c = 314;             // [kg/m3]
+		const double M = 28.013;
+		const double k = 1.38062e-23;
+		const double eps = 138.08483e-23;
+		const double N_A = 6.02213E26;
+		const double R = 8.31434;
```

Comment:
When starting this in OGS6, I tried to be as close as possible to OGS5. I'd like to fix this within a general cleanup. Btw: Now there are some clean physical constants files in OGS5, they could be copied to OGS6, too.